src/coreclr/vm/frames.h

```diff
@@ -355,19 +254,17 @@ typedef DPTR(FrameTypeName) PTR_FrameTypeName;
 
 class FrameBase
```

Comment:
You can delete FrameBase and fold it into Frame. FrameBase

Also, I think you can delete the machinery that adds GSCookies in front of the Frame. GSCookies in front of the frame existed solely to protect the vtable pointer. (One can inject arbitrary control flow by hacking vtable pointer - that's no longer possible when the control flow is done via switch statements.)